program/src/instructions/compressed/buy.rs

```diff
@@ -151,10 +153,12 @@ pub fn process_buy<'info>(
     let list_state = &ctx.accounts.list_state;
 
     // In case we have an extra remaining account.
+    // TODO: no checked math?
     let mut v = Vec::with_capacity(ctx.remaining_accounts.len() + 1);
 
     // Validate the cosigner and fetch additional remaining account if it exists.
     // Cosigner could be a remaining account from an old client.
+    // TODO: I really dont get what we're trying to do here by stuffing cosigner into remaining accounts?
```

Comment:
This is an ugly hack to make this handler backwards compatible with the old client/IDL.